cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -325,6 +325,7 @@ pub enum TypedManifest {
 /// Modifications should be made using the the raw functions in this module.
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
 #[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+#[serde(deny_unknown_fields)]
```

Comment:
Nit, non-blocking: the user-facing error that this translates into is slightly misleading because we know it's valid TOML. It's just not the TOML that we want 😄 
```
% just flox edit
make: Entering directory '/Users/dcarley/projects/flox/flox/pkgdb'
mkdir -p tests;
Generating tests/.gitignore
make: Leaving directory '/Users/dcarley/projects/flox/flox/pkgdb'
~/projects/flox/flox/cli ~/projects/flox/flox
❌ ERROR: Failed to parse manifest: unknown field `blah`, expected one of `version`, `install`, `vars`, `hook`, `profile`, `options`



Please ensure that '.flox/env/manifest.toml' is a valid TOML file.
```

Reply:
ah that's separate from the parsing, but agree we could tweak that, although i don't think we actually _know_ whether it's invalid toml or an invalid schema.



I wonder what that looked like before/whether this is a regression?